src/librbd/cache/rwl/LogEntry.cc

```diff
@@ -73,6 +73,7 @@ std::ostream &operator<<(std::ostream &os,
   return entry.format(os);
 }
 
+#ifdef WITH_RBD_RWL
```

Comment:
Wouldn't you need a `init` method definition?

Reply:
Yes it's needed. This is refactored in the next PR (which includes aio_write workflow)

Reply:
Let's make sure that a given PR can compile, link, and run all by itself and not require a future PR to fix any build issues.

Reply:
I'm not sure I understand - this PR does not have any build issues when built with WITH_RBD_RWL=ON and WITH_RBD_SSD=ON. This PR only has the framework in place for SSD and RWL to share code. Would you like me to push the update to have at least `init` implemented for SSD?

Reply:
You can't just push a partially implemented PR -- it needs to be able to stand by itself. Perhaps it would make more sense to start with a PR that just moves the existing RWL cache code to new directories/namespaces/etc and introduce the new common base class. In the next PR you can start to implement the SSD cache.